pkg/localclusterimport/import_local_cluster.go

```diff
@@ -0,0 +1,262 @@
+package localclusterimport
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	hiveext "github.com/openshift/assisted-service/api/hiveextension/v1beta1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/openshift/hive/apis/hive/v1/agent"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Local_cluster_import struct {
+	clusterImportOperations ClusterImportOperations
+	localClusterNamespaceName    string
+	log                          *logrus.Logger
+}
+
+func NewImportLocalCluster(localClusterImportOperations ClusterImportOperations, log *logrus.Logger, localClusterNamespaceName string) Local_cluster_import {
+	return Local_cluster_import{clusterImportOperations: localClusterImportOperations, log: log, localClusterNamespaceName: localClusterNamespaceName}
+}
+
+func (i *Local_cluster_import) shouldImportLocalCluster() (bool, error) {
```

Comment:
what will happen after upgrade? will this cluster be usable? or do you need to re-create it?

Reply:
That's a good question, how do we know an upgrade has happened?

Reply:
probably a different cluster version.